MetricsUploader/include/MetricsUploader/MetricsUploader.h

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_METRICS_UPLOADER_METRICS_UPLOADER_H_
+#define ORBIT_METRICS_UPLOADER_METRICS_UPLOADER_H_
+
+#ifdef _WIN32
+#include <windows.h>
+#endif  // WIN32
+
+#include <absl/synchronization/mutex.h>
+
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <string_view>
+
+#include "MetricsUploader/Result.h"
+#include "OrbitBase/Result.h"
+#include "orbit_log_event.pb.h"
+
+namespace orbit_metrics_uploader {
+
+constexpr std::string_view kMetricsUploaderClientDllName = "metrics_uploader_client";
+
+// This class is used for sending log events from Orbit. It works only on Windows and only if
+// metrics_uploader_client.dll is available. The types of logs that could be send are defined in
+// orbit_client_protos::OrbitLogEvent::LogEventType.
+//
+// Only one instance of the class is allowed.
+//
+// Usage example:
+//
+// auto metrics_uploader = MetricsUploader::CreateMetricsUploader();
+// if (metrics_uploader.has_value()) {
+//   metrics_uploader.value()->SendLogEvent(...);
+// }
+//
+class MetricsUploader {
+ public:
+  // Create a MetricsUploader instance, load metrics uploader client library if available and starts
+  // metrics uploader client when called first time. Return the instance if there are no errors and
+  // ErrorMessage otherwise.
+  [[nodiscard]] static ErrorMessageOr<std::unique_ptr<MetricsUploader>> CreateMetricsUploader(
+      std::string_view client_name = kMetricsUploaderClientDllName);
```

Comment:
When a move constructor is defined (see instructions below) you could return a `ErrorMessageOr<MetricsUploader>` and avoid the heap allocation. To me this is a better design, but I don't feel particularly strong about this.

Reply:
Agree, removed `unique_ptr` here.